src/components/menus/ExtensionMenu.svelte

```diff
@@ -0,0 +1,68 @@
+<svelte:options accessors={true} immutable={true} />
+
+<script lang="ts">
+  import SettingsIcon from '@nasa-jpl/stellar/icons/settings.svg?component';
+  import { createEventDispatcher } from 'svelte';
+  import type { User } from '../../types/app';
+  import type { Extension } from '../../types/extension';
+  import { permissionHandler } from '../../utilities/permissionHandler';
+  import PlanNavButton from '../plan/PlanNavButton.svelte';
+  import MenuItem from './MenuItem.svelte';
+
+  export let extensions: Extension[];
+  export let user: User | null;
+
+  const dispatch = createEventDispatcher();
+
+  function callExtension(extension: Extension) {
+    dispatch('callExtension', extension);
+  }
+
+  function hasExtensionPermission(extension: Extension): boolean {
+    for (const extensionRole of extension.extension_roles) {
+      if (user?.activeRole === extensionRole.role) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+</script>
+
+{#if extensions.length > 0}
+  <div class="view-menu st-typeography-medium">
+    <PlanNavButton title="Extensions" menuTitle="">
+      <SettingsIcon />
+      <div class="view-menu st-typography-medium" slot="menu">
+        {#each extensions as extension}
+          <MenuItem
+            on:click={() => callExtension(extension)}
+            use={[
+              [
+                permissionHandler,
+                {
+                  hasPermission: hasExtensionPermission(extension),
+                  permissionError: 'You do not have permission to call this extension',
+                },
+              ],
+            ]}
+          >
+            {extension.label}
```

Comment:
Could also add the extension description in a line below the label?

Reply:
Perhaps just showing it below the label with a max char limit + ellipses? Showing it on hover may result in the list items jumping around a lot since the list item heights would change dynamically.

Reply:
Truncation looks good, one last thing would be a bit more styling for the description, perhaps something like:
```JSX
<div class="extension-menu--menu-item">
   {extension.label}
   <span class="st-typography-label">{formatDescription(extension.description)}</span>
</div>
```

and 

```CSS
.extension-menu--menu-item {
    display: flex;
    flex-direction: column;
  }
```

Reply:
Which looks something like this 
<img width="363" alt="image" src="https://github.com/NASA-AMMOS/aerie-ui/assets/4419607/dba7c22e-c8dc-4173-96d2-f139812c458d">

Reply:
Yeah that looks great, I'll make this change!